### ISSUE TITLE
why there is no sampler info in reflection information

### ISSUE THREAD

**Issue by everywill** [brendan-duncan/wgsl_reflect#51]
on the [example](https://brendan-duncan.github.io/wgsl_reflect/example.html) page i tried a shader with sampler but there is no sampler info in the reflection  information.
here is the shader:

struct VertexOutput {
    @builtin(position) Position: vec4<f32>,
    @location(0) attr_var_UV: vec2<f32>
}

@binding(1) @group(1) var<uniform> color: texture_2d<f32>;
@binding(0) @group(1) var tex_sprite: sampler;

@fragment
fn fragment(fragData: VertexOutput) -> @location(0) vec4f {
    var _33: vec4<f32> = textureSample(color, tex_sprite, fragData.attr_var_UV);
    var _34: f32 = _33.x;
    var _41: f32 = _33.y;
    var _48: f32 = _33.z;
    var _57: vec3<f32> = vec3<f32>(pow((_34 + 0.054999999701976776123046875) * 0.947867333889007568359375, 2.400000095367431640625), _41, _48) * _33.w;
    return vec4<f32>(_57.x, _57.y, _57.z, _33.w);
}

and the reflection info

TIME: 1.2000000476837158ms
Entry Points
fragment: fragment
Inputs
Position: vec4<f32> location: position
attr_var_UV: vec2<f32> location: 0
Uniforms
color: texture_2d
Buffer Size: 0
Bind Group: 1
Bind Index: 1
Members:

I wonder if there should be the sampler info or not?

**Comment by brendan-duncan** [comment]
To provide a public answer to this:
1. The example is quite simplistic, I figured people are primarily interested in uniform buffer layouts.
2. The shader you posted does have an error, texture_2d needs to have a sampler type, as in texture_2d<f32>.

I updated the example to list textures and samplers. I also have it catch errors and print the error if it fails to parse the shader. In this case it now tells you "Expected '<' for sampler type. Line 6. Found: ';'."
